arroyo-worker/src/connectors/kafka/source/mod.rs

```diff
@@ -207,12 +208,14 @@ where
                         .get(&p.id())
                         .map(|s| Offset::Offset(s.offset))
                         .unwrap_or_else(|| {
-                            if has_state {
+                            if offset_mode == Offset::Stored {
+                                offset_mode
```

Comment:
I think that this should still go into the else block here. If `has_state` is true, that means that we are restoring an existing checkpoint and we should now be managing offsets based on our state. If we find a partition that doesn't exist in our state, that means it was added since our last checkpoint. In that case, I think the correct behavior is to always read it from the beginning.

Reply:
apologies, this was an oversight on my part. I was thinking from the consumer point of view where I would just want to start from an x offset but the correct way would be to re-create the pipeline itself if offset differs from checkpoint.